packages/push/src/impl/PushRegistrationWebpushImpl.ts

```diff
@@ -0,0 +1,141 @@
+import {AbstractPushRegistration} from "../AbstractPushRegistration";
+import {ConfigurationService, ServiceConfiguration} from "@aerogear/core";
+import {PushRegistrationOptions, PushRegistrationWebpushOptions} from "../PushRegistration";
+
+declare var window: any;
+
+const DEFAULT_SERVICE_WORKER: string = "/service-worker.js";
+
+/**
+ * AeroGear UPS registration SDK - Webpush implementation
+ *
+ * Usage:
+ * // Initialize SDK first
+ * app.init(config);
+ * let registration = new PushRegistration();
+ * registration.register("myAppleOrFirebaseToken");
+ */
+export class PushRegistrationWebpushImpl extends AbstractPushRegistration {
+  constructor(config: ConfigurationService) {
+    super(config);
+    // The config is valid
+  }
+
+  public async register(options: PushRegistrationWebpushOptions = {}): Promise<void> {
+    const {alias, categories, serviceWorker} = options;
+
+    if (this.validationError) {
+      throw new Error(this.validationError);
+    }
+
+    if (this.httpClient) {
+      const subscription = await this.subscribeToWebPush(this.platformConfig.appServerKey, serviceWorker);
+      const postData = {
+        "deviceToken": JSON.stringify(subscription),
+        "deviceType": "ChromeBrowser",
```

Comment:
Maybe we should think in a way to get the browser name

Reply:
There's `window.navigator.userAgent` which may be useful.